client/components/status/status.js

```diff
@@ -1,55 +1,178 @@
 import "./index.css";
+import "react-toastify/dist/ReactToastify.css";
 
 import axios from "axios";
 import PropTypes from "prop-types";
 import qs from "qs";
 import React from "react";
-import { Cookies } from "react-cookie";
-import { Link } from 'react-router-dom';
-import { toast } from "react-toastify";
-import 'react-toastify/dist/ReactToastify.css';
-
-import { validateApiUrl, loginError, logoutSuccess, mainToastId } from "../../constants";
+import {Cookies} from "react-cookie";
+import {Link} from "react-router-dom";
+import {toast} from "react-toastify";
+import {
+  genericError,
+  getUserRadiusSessionsUrl,
+  logoutSuccess,
+  mainToastId,
+  validateApiUrl,
+} from "../../constants";
 import getText from "../../utils/get-text";
+import LoadingContext from "../../utils/loading-context";
 import logError from "../../utils/log-error";
 import Contact from "../contact-box";
 
 export default class Status extends React.Component {
-  componentDidMount() {
-    const { cookies, orgSlug, logout } = this.props;
-    const token = cookies.get(`${orgSlug}_auth_token`);
-    const url = validateApiUrl.replace("{orgSlug}", orgSlug);
-    return axios({
-      method: "post",
-      headers: {
-        "content-type": "application/x-www-form-urlencoded",
-      },
-      url,
-      data: qs.stringify({
-        token,
-      }),
-    })
-      .then(response => {
-        if (response.data["control:Auth-Type"] !== "Accept") {
+  constructor(props) {
+    super(props);
+    this.loginIfameRef = React.createRef();
+    this.loginFormRef = React.createRef();
+    this.logoutIfameRef = React.createRef();
+    this.logoutFormRef = React.createRef();
+    this.state = {
+      username: "",
+      password: "",
+      sessions: [],
+    };
+    this.validateToken = this.validateToken.bind(this);
+    this.getUserRadiusSessions = this.getUserRadiusSessions.bind(this);
+  }
+
+  async componentDidMount() {
+    const {cookies, orgSlug} = this.props;
+    // to prevent recursive call in case redirect url is status page
+    if (window.top === window.self) {
+      const {sessions} = this.state;
+      const isValid = await this.validateToken();
+      if (isValid) {
+        const macaddr = cookies.get(`${orgSlug}_macaddr`);
+        if (macaddr) {
+          await this.getUserRadiusSessions();
+          if (sessions && sessions.length === 0) {
```

Comment:
I think we should add a comment above this to explain we'll do the log in only if there's no other active session.
BTW I think we should use the mac address param (if not empty) here to filter the sessions, because otherwise users may not be able to log in with two different devices using the same account

Reply:
@nemesisdesign  Since mac address is stored in a cookie, it's already available to the proxy server and there was no need to send it from front-end. I've already added this filter to [proxy server's code](https://github.com/openwisp/openwisp-wifi-login-pages/blob/432e88b4c7913b129f5b30835535c965a86c3d47/server/controllers/user-radius-sessions-controller.js#L23)
And I'll add the comment explaining that.